setup.py

```diff
@@ -79,6 +79,9 @@
 #   USE_LMDB
 #     enables use of LMDB for storage
 #
+#   USE_PROTOTYPES
```

Comment:
I think we're going by BUILD_* nowadays.

Reply:
err, no I don't think so? cc @xuhdev

Reply:
oh, maybe for features

Reply:
I'm not sure? @xuhdev changed USE_NAMEDTENSOR to BUILD_NAMEDTENSOR

Reply:
(I don't think any of my opinion below has been explicitly written down as rules; It is based on my experience of common practice and what PyTorch has already done.)

The change from `USE_NAMEDTENSOR` to `BUILD_NAMEDTENSOR` is because (consistently with other build options) named tensor is a purely new feature to **build**. `USE_` options are more towards adding a new dependency, such as `USE_FBGEMM`. In other words, semantically,  `BUILD_` leans more towards adding a feature largely implemented in PyTorch, and `USE_` leans more towards adding a new backend library. If some build option is in between (implement a lot in PyTorch but also add a giant new dependency), I would say one can choose the best naming based on context and what similar build options do.

IMO a rule of thumb to use `USE_` is to ask whether this option can be simply documented as "enable the use of [some library]". But, of course, consistency with other build options also needs to be taken into consideration.